tooltester/src/test/java/io/dockstore/tooltester/client/cli/ClientTest.java

```diff
@@ -1,18 +1,263 @@
-        OptionParser parser = new OptionParser();
-        final OptionSet parsed = parser.parse("");
-        Client client = new Client(parsed);
+        client = new Client();
         client.setupClientEnvironment();
         Assert.assertTrue("client API could not start", client.getContainersApi() != null);
     }
 
+    /**
+     * Tests if Jenkins server is set up
+     */
+    @Test
+    public void setupJenkins() {
+        client.setupJenkins();
+        Assert.assertTrue("Jenkins server can not be reached", client.getJenkins() != null);
+    }
+
+    /**
+     * This is for admin use only.  It deletes all created Jenkins pipelines
+     */
+   private void deleteJenkinsTests() {
+        client.setupJenkins();
+        JenkinsServer jenkins = client.getJenkins();
+        Assert.assertTrue("Jenkins server can not be reached", jenkins != null);
+        client.deleteJobs("DockerfileTest");
+        client.deleteJobs("ParameterFileTest");
+        client.deleteJobs("PipelineTest");
+    }
+
+    /**
+     * Test with unknown command
+     */
+    @Test
+    public void unknownCommand() {
+        String[] argv = { "mmmrrrggglll" };
+        exit.expectSystemExitWithStatus(COMMAND_ERROR);
+        main(argv);
+    }
+
+    /**
+     * Test enqueue with no parameters which should print the help usage
+     */
+    @Test
+    public void enqueue() {
+        String[] argv = { "enqueue" };
+        main(argv);
+    }
+
+    /**
+     * Test enqueue with option --all which should run all jobs
+     */
+    @Test
+    public void enqueueAll() {
+        String[] argv = { "enqueue" , "--all"};
+        main(argv);
+    }
+
+    /**
+     * Test enqueue with default options
+     */
+    @Test
+    public void enqueueTool() {
+        String[] argv = { "enqueue", "--tool" , "quay.io/pancancer/pcawg-bwa-mem-workflow"};
+        main(argv);
+    }
+
+    /**
+     * This tests the Jenkins pipeline creation
+     */
+    @Test
+    public void createJenkinsTests() {
+        String[] argv = { "--execution", "local", "--source", "Docktesters group", "--api", "https://www.dockstore.org:8443/api/ga4gh/v1" };
+        main(argv);
+    }
+
+    /**
+     * This tests the client with no parameters
+     */
+    @Test
+    public void empty() {
+        String[] argv = {""};
+        main(argv);
+    }
+
+    /**
+     * This gets the report of all the tools
+     */
+    @Test
+    public void report() {
+        String[] argv = { "report" };
+        main(argv);
+    }
+
+    /**
+     * This reports on a specific tool
+     */
+    @Test
+    public void reportTool() {
+        String[] argv = new String[] { "report", "--tool", "quay.io/pancancer/pcawg-bwa-mem-workflow" };
+        main(argv);
+    }
+
+    /**
+     * This reports on specific tools
+     */
+    @Test
+    public void reportTools() {
+        String[] argv = new String[] { "report", "--tool", "quay.io/pancancer/pcawg-bwa-mem-workflow" };
+        main(argv);
```

Comment:
These tests don't really assert anything.
Start checking that a report was created, where, etc.